File: __tests__/fixtures/java/DeadCodeUnusedMethods.java
package com.example;

public class DeadCodeUnusedMethods {

    private void usedMethod() {
        System.out.println("used");
    }

    private void unusedMethod() {
        System.out.println("unused");
    }

    private void usedViaReference() {
        System.out.println("ref");
    }

    public void publicMethod() {
        System.out.println("public");
    }

    protected void protectedMethod() {
        System.out.println("protected");
    }

    public void process() {
        usedMethod();
        Runnable r = this::usedViaReference;
    }
}

File: __tests__/fixtures/kotlin/DeadCodeUnusedMethods.kt
package com.example

class DeadCodeUnusedMethods {

    private fun usedMethod() {
        println("used")
    }

    private fun unusedMethod() {
        println("unused")
    }

    private fun usedViaReference() {
        println("ref")
    }

    fun publicMethod() {
        println("public")
    }

    fun process() {
        usedMethod()
        val ref = ::usedViaReference
    }
}
